File: go-gin-crud/app/model/baseModel.go
package model

import (
	"github.com/dgrijalva/jwt-go"
	"gorm.io/gorm/logger"
)

type SqlLogger struct {
	logger.Interface
}
type tokenClaims struct {
	Id string `json:"id"`
	jwt.StandardClaims
}
type Pagination struct {
	Page     int `json:"page"`
	PageSize int `json:"pageSize"`
	// Total    int `json:"total"`
}
type PaginationResponse struct {
	RecordPerPage int `json:"recordPerPage"`
	CurrentPage   int `json:"currentPage"`
	TotalPage     int `json:"totalPage"`
	TotalRecord   int `json:"totalRecord"`
}
type ValidatorErr struct {
	Field   string      `json:"field"`
	Value   interface{} `json:"value"`
	Tag     string      `json:"tag"`
	Param   string      `json:"param"`
	Type    string      `json:"type"`
	Message string      `json:"message"`
}

File: beginner/frontend/vue-ts-frontend/src/models/base.model.ts
export interface IResponseData {
  data: any[];
  pagination: IPagination;
}

export interface IResponse {
  data: any;
}

export interface IPagination {
  currentPage: number;
  recordPerPage: number;
  totalPage: number;
  totalRecord: number;
}

export interface IError {
  code: string;
  message: string;
  validator?: IValidator[];
}

interface IValidator {
  field: string;
  value: string;
  tag: string;
  type: string;
  param: string;
  message: string;
}
